File: SDKs/amazonaws.com/qldb-session/2019-07-11/java/lib/src/main/java/openapisdk/models/shared/ExecuteStatementRequest.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * ExecuteStatementRequest
 * Specifies a request to execute a statement.
**/
public class ExecuteStatementRequest {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Parameters")
    public ValueHolder[] parameters;
    public ExecuteStatementRequest withParameters(ValueHolder[] parameters) {
        this.parameters = parameters;
        return this;
    }
    @JsonProperty("Statement")
    public String statement;
    public ExecuteStatementRequest withStatement(String statement) {
        this.statement = statement;
        return this;
    }
    @JsonProperty("TransactionId")
    public String transactionId;
    public ExecuteStatementRequest withTransactionId(String transactionId) {
        this.transactionId = transactionId;
        return this;
    }
}

File: SDKs/amazonaws.com/qldb-session/2019-07-11/python/src/sdk/models/shared/executestatementrequest.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import valueholder as shared_valueholder


@dataclass_json
@dataclasses.dataclass
class ExecuteStatementRequest:
    r"""ExecuteStatementRequest
    Specifies a request to execute a statement.
    """
    
    statement: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('Statement') }})
    transaction_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('TransactionId') }})
    parameters: Optional[list[shared_valueholder.ValueHolder]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Parameters') }})
    
